src/main/java/net/fabricmc/tinyremapper/TinyRemapper.java

```diff
@@ -206,11 +220,13 @@ private TinyRemapper(Collection<IMappingProvider> mappingProviders, boolean igno
 			boolean rebuildSourceFilenames,
 			boolean skipLocalMapping,
 			boolean renameInvalidLocals,
-			ClassVisitor extraAnalyzeVisitor, Remapper extraRemapper) {
+			ClassVisitor extraAnalyzeVisitor, Remapper extraRemapper, WrapperFunction pre, WrapperFunction post) {
 		this.mappingProviders = mappingProviders;
 		this.ignoreFieldDesc = ignoreFieldDesc;
 		this.threadCount = threadCount > 0 ? threadCount : Math.max(Runtime.getRuntime().availableProcessors(), 2);
 		this.keepInputData = keepInputData;
+		this.pre = pre;
+		this.post = post;
```

Comment:
Why would you only allow one pre/post visitor. Imagine that there could be different pre/post visitors for different purposes and written by different people, they should all be able to apply in sequence. For example, it should allow `builder.extraPre(...).extraPre(...).extraPre(...)` and apply them in sequence.

Reply:
`ClassVisitor`s naturally chain, so if you had multiple you wanted to apply you would simply chain them in your `WrapperFunction`

Reply:
But then the function name should not be called `extraPre` or `extraPost` as it confuses people. It should be `setPre` or `setPost` since the "extra" indicates you can have them chain together.

Reply:
the other methods don't follow that convention